cmd/query/app/static_handler.go

```diff
@@ -44,24 +46,30 @@ func RegisterStaticHandler(r *mux.Router, logger *zap.Logger, qOpts *QueryOption
 	staticHandler, err := NewStaticAssetsHandler(qOpts.StaticAssets, StaticAssetsHandlerOptions{
 		BasePath:     qOpts.BasePath,
 		UIConfigPath: qOpts.UIConfig,
+		Logger:       logger,
 	})
+
 	if err != nil {
 		logger.Panic("Could not create static assets handler", zap.Error(err))
 	}
+
 	staticHandler.RegisterRoutes(r)
 }
 
 // StaticAssetsHandler handles static assets
 type StaticAssetsHandler struct {
-	options   StaticAssetsHandlerOptions
-	indexHTML []byte
-	assetsFS  http.FileSystem
+	options      StaticAssetsHandlerOptions
+	indexHTML    []byte
+	assetsFS     http.FileSystem
+	watching     *sync.WaitGroup
+	contentMutex sync.Mutex
```

Comment:
rather than using a mutex, we can use https://golang.org/pkg/sync/atomic/#Value with Store/Load